notebooks/config.ipynb

```diff
@@ -0,0 +1,1046 @@
+      "File \u001b[0;32m~/NOAA/uwtools/src/uwtools/config/tools.py:107\u001b[0m, in \u001b[0;36mrealize_config\u001b[0;34m(input_config, input_format, update_config, update_format, output_file, output_format, key_path, values_needed, total, dry_run)\u001b[0m\n\u001b[1;32m    105\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m {}\n\u001b[1;32m    106\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m total \u001b[38;5;129;01mand\u001b[39;00m unrendered(\u001b[38;5;28mstr\u001b[39m(input_obj)):\n\u001b[0;32m--> 107\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m UWConfigRealizeError(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mConfig could not be totally realized\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    108\u001b[0m output_class \u001b[38;5;241m=\u001b[39m format_to_config(output_format)\n\u001b[1;32m    109\u001b[0m output_class\u001b[38;5;241m.\u001b[39mdump_dict(cfg\u001b[38;5;241m=\u001b[39moutput_data, path\u001b[38;5;241m=\u001b[39moutput_file)\n",
+      "\u001b[0;31mUWConfigRealizeError\u001b[0m: Config could not be totally realized"
```

Comment:
I think it would be helpful to add a comment that the reason this error message comes up is because the `id` key-value pair was not rendered since the user didn't supply the value. The idea gets a little lost since the error message doesn't formally say that `id` was the key that could not be rendered and just says `Config could not be totally realized`.

The next example is helpful to see once `id` is added back in that it can now fully realize the config. Either adding a comment before or after would be great.